### PR TITLE
Enhances array literals with meaningful brackets.

### PR DESCRIPTION
**Pull request by christianrolle**
Change the style guide in terms of using square brackets for Array literals instead of parentheses for expressiveness reasons (I think percent literal delimiters should be meaningful). Square brackets reveal the intention, that the literal creates an Array, like:

~~~ruby
# good
%w[one two three]
%i[one two three]
~~~

### CODE REVIEW

**Comment by exAspArk** [comment]
:+1:

**Comment by linchus** [comment]
:+1:

**Comment by 0x01f7** [comment]
I don't think it's more meaningful... :confused:

**Comment by christianrolle** [comment]
What do you expect from something like "[]" to generate, compared to "()"?
In Ruby "[]" is very well know for accessing/creating a collection, don't you agree?
So why not using that practice for creating Arrays by percent literals?
I mean in Ruby we create an Array like:

~~~ruby
numbers = [1, 2, 3]
~~~

so why not leaning on it:

~~~ruby
names = %w[linchus exAspArk 0x01f7]
~~~

To me that is consistent.

File: README.md [hunk 1]
```diff
@@ -3627,19 +3627,20 @@ resource cleanup when possible.
 <sup>[[link](#percent-s)]</sup>
 
 * <a name="percent-literal-braces"></a>
-  Prefer `()` as delimiters for all `%` literals, except `%r`. Since parentheses
-  often appear inside regular expressions in many scenarios a less common
-  character like `{` might be a better choice for a delimiter, depending on the
-  regexp's content.
+  Prefer `()` as delimiters for all `%` literals, except `%r` and Array
```
Comment (bbatsov) [inline README.md]: Just a tiny remark - should Array be capitalised? I guess it depends on whether we're discussing the class or arrays in general.

**Comment by bbatsov** [comment]
Please change the commit message from `Enhances Array literals with meaningful brackets.` to `Enhance array literals with meaningful brackets`.

**Comment by christianrolle** [comment]
good point.

**Comment by christianrolle** [comment]
Done (#532).

**Comment by bbatsov** [comment]
Good. Just squash those 3 commits together and kill the `.` from the title of the resulting commit to match our conventions.

**Comment by 0x01f7** [comment]
> What do you expect from something like "[]" to generate, compared to "()"?
> In Ruby "[]" is very well know for accessing/creating a collection, don't you agree?
> So why not using that practice for creating Arrays by percent literals?

I know it shoud be an array literal, because I see `%w`, neither `[]`, nor `()`. So both are OK for me. But use `()` as delimiters for **all `%` literals**, that is consistent to me.

Of course, In Ruby `[]` is very well know for accessing/creating a collection. But if we change the rule, what does **all `%` literals** mean?

+ %x, %s ----> avoid use
+ %r ----> use `{}`
+ %q, %Q ----> use `()`
+ %w, %W, %i, %I ----> use `[]`

I don't mind we use `()` or `[]` for array literals. But this modification make the rule unclear.

**Comment by christianrolle** [comment]
Yeah. I totally agree. That's why I first wanted to remove the **all** `%` **literals** rule.
The rule already is inconsistent, since `{}` have to be preferred for regexp.

>  %r ----> use `{}`

So the Ruby style guide already does **not** suggest to use `()` for all literals.
To me a solution would be removing the recommendation to prefer `()` for all literals (since curly braces are obviously no `()`).

**Comment by 0x01f7** [comment]
I agree with you. The rule should be rewritten. By the way, don't forget modify the sample code in other rules([single-line-blocks](https://github.com/bbatsov/ruby-style-guide#single-line-blocks), [array-join](https://github.com/bbatsov/ruby-style-guide#array-join), [percent-w](https://github.com/bbatsov/ruby-style-guide#percent-w), etc.).

**Comment by christianrolle** [comment]
Good point.
Maybe some others have some opinions about rewriting the rule as 0x01f7 and me suggest?

**Comment by exAspArk** [comment]
Hi, so you are suggesting to stop using `[]` or `{}` for `%` literals? And always to use `()`?

IMHO, unifying the guide is a good idea. However, it's not so clear for me why we should prefer `()` over other characters? Is there any reasons? Why not always to use `{}` then? 
Or maybe make it not mandatory?

**Comment by exAspArk** [comment]
Hi! So you are suggesting to remove the guide? :smile_cat: Or leave just a part about `{}` for `%r` literals? 

What do you think about unifying the guide? Why not to always use `{}` then?

**Comment by christianrolle** [comment]
@exAspArk: I guess there is some misunderstanding. I suggest to rephrase it.
My intention was to increase expressiveness by using ´[]´ braces for array literals.

**Comment by 0x01f7** [comment]
> Hi! So you are suggesting to remove the guide? :smile_cat: Or leave just a part about {} for %r literals? What do you think about unifying the guide?

I suggest to
 
* Keep the original version
* Or accept this PR, but rephrase it

Both are OK for me. I am worried about the modification makes the rule too complex, and hard to understand.

> Why not to always use {} then?

Of course you can. It's all your choice.

**Comment by christianrolle** [comment]
I close this pull request as not approved, since there seem to some reservations against it.

**Pull request closed without merging**